operations/helm/charts/mimir-distributed/templates/query-scheduler/query-scheduler-dep.yaml

```
@@ -0,0 +1,117 @@
+  annotations:
+    {{- toYaml .Values.query_scheduler.annotations | nindent 4 }}
+spec:
+  replicas: {{ .Values.query_scheduler.replicas }}
+  selector:
+    matchLabels:
+      {{- include "mimir.selectorLabels" (dict "ctx" . "component" "query-scheduler") | nindent 6 }}
+  strategy:
+    {{- toYaml .Values.query_scheduler.strategy | nindent 4 }}
+  template:
+    metadata:
+      labels:
+        {{- include "mimir.podLabels" (dict "ctx" . "component" "query-scheduler") | nindent 8 }}
+        {{- with .Values.query_scheduler.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      annotations:
+        {{- include "mimir.podAnnotations" (dict "ctx" . "component" "query-scheduler") | nindent 8 }}
+    spec:
+      serviceAccountName: {{ template "mimir.serviceAccountName" . }}
+    {{- if .Values.query_scheduler.priorityClassName }}
+      priorityClassName: {{ .Values.query_scheduler.priorityClassName }}
+    {{- end }}
+      securityContext:
+        {{- toYaml .Values.query_scheduler.securityContext | nindent 8 }}
+      initContainers:
+        {{- toYaml .Values.query_scheduler.initContainers | nindent 8 }}
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
+      containers:
+        - name: query-scheduler
+          image: "{{ include "mimir.imageReference" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - "-target=query-scheduler"
+            - "-config.expand-env=true"
+            - "-config.file=/etc/mimir/mimir.yaml"
+          {{- range $key, $value := .Values.query_scheduler.extraArgs }}
+            - "-{{ $key }}={{ $value }}"
+          {{- end }}
+          volumeMounts:
+            {{- if .Values.query_scheduler.extraVolumeMounts }}
+              {{ toYaml .Values.query_scheduler.extraVolumeMounts | nindent 12}}
+            {{- end }}
+            - name: runtime-config
+              mountPath: /var/{{ include "mimir.name" . }}
+            {{- if .Values.enterprise.enabled }}
+            - name: license
+              mountPath: /license
+            {{- end }}
+            - name: config
+              mountPath: /etc/mimir
+          ports:
+            - name: http-metrics
+              containerPort: {{ include "mimir.serverHttpListenPort" . }}
+              protocol: TCP
+            - name: grpc
+              containerPort: {{ include "mimir.serverGrpcListenPort" . }}
+              protocol: TCP
+          livenessProbe:
+            {{- toYaml .Values.query_scheduler.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml .Values.query_scheduler.readinessProbe | nindent 12 }}
+          resources:
+            {{- toYaml .Values.query_scheduler.resources | nindent 12 }}
+          securityContext:
+            readOnlyRootFilesystem: true
```

Following what other components have
```suggestion
            {{- toYaml .Values.query_scheduler.containerSecurityContext | nindent 12 }}
```